### PR TITLE
Improve and streamline plotting 

### PR DESCRIPTION
**Pull request by mdmurbach**
- [x] Add option for interactive Altair plot
- [x] Add Bode plot option
- [x] Streamline `circuit.plot()` method to generate Nyquist, Bode, or Altair plots via a consistent use of `plotting.plot_*()` functions
- [x] Update example docs to demonstrate how to use new plots

### CODE REVIEW

File: impedance/plotting.py [hunk 1]
```diff
@@ -16,8 +18,7 @@ def _set_orderOfMagnitude(self, range):
 
 
 def plot_nyquist(ax, Z, scale=1, units='Ohms', fmt='.-', **kwargs):
```
Comment (BGerwe) [inline impedance/plotting.py]: What is the expected behavior of "scale" it doesn't seem to change anything?

File: impedance/plotting.py [hunk 2]
```diff
@@ -73,3 +74,196 @@ def plot_nyquist(ax, Z, scale=1, units='Ohms', fmt='.-', **kwargs):
...
+
+        Returns
+        -------
+        ax: matplotlib.axes.Axes
+    """
+
+    ax_mag, ax_phs = axes
+
+    ax_mag.plot(f, np.abs(Z), fmt, **kwargs)
+    ax_phs.plot(f, -np.angle(Z), fmt, **kwargs)
```
Comment (BGerwe) [inline impedance/plotting.py]: I think we want the phase to show in degrees, not radians.

`ax_phs.plot(f, -np.angle(Z, deg=True), fmt, **kwargs)`

File: impedance/plotting.py [hunk 3]
```diff
@@ -73,3 +74,196 @@ def plot_nyquist(ax, Z, scale=1, units='Ohms', fmt='.-', **kwargs):
...
+        bode = alt.Chart(df).mark_line().encode(
+            alt.X('f:Q', axis=alt.Axis(title="f [Hz]"),
+                  scale=alt.Scale(type='log', nice=False)),
+            color='kind:N'
+        ).properties(
+            width=size,
+            height=size/2 - 25
+        ).transform_calculate(
+            mag="sqrt(pow(datum.z_real,2) + pow(datum.z_imag,2))",
+            neg_phase="-atan(datum.z_imag/datum.z_real)"
```
Comment (BGerwe) [inline impedance/plotting.py]: Also check to see if this is degrees or radians

File: impedance/plotting.py [hunk 4]
```diff
@@ -73,3 +74,196 @@ def plot_nyquist(ax, Z, scale=1, units='Ohms', fmt='.-', **kwargs):
...
+            size=alt.condition(nearest, alt.value(80), alt.value(30)),
+            color='kind:N'
+        ).add_selection(
+            nearest
+        ).properties(
+            width=size,
+            height=size/2 - 25
+        ).transform_calculate(
+            mag="sqrt(pow(datum.z_real,2) + pow(datum.z_imag,2))",
+            neg_phase="-atan(datum.z_imag/datum.z_real)"
```
Comment (BGerwe) [inline impedance/plotting.py]: Also check if degrees or radians.

**Comment by BGerwe** [comment]
This is awesome! One thing I noticed is Altair doesn't put a white background behind the art so people using the Dark theme in Jupyter can't see any text. 
image

File: impedance/plotting.py (hunk 1 above)
Comment (mdmurbach) [reply impedance/plotting.py]: In the past it set the formatting of the x and y axes (to show 10^3 for example and fixing #12) in these lines     https://github.com/ECSHackWeek/impedance.py/blob/8a29e65099279fc489f1f54892327dee67328e50/impedance/plotting.py#L67-L73

Not sure why it's not working any more...

File: impedance/plotting.py (hunk 2 above)
Comment (mdmurbach) [reply impedance/plotting.py]: Good catch :) I'll fix this with a new commit

File: impedance/plotting.py (hunk 1 above)
Comment (mdmurbach) [reply impedance/plotting.py]: The work around to get this to work originally is here: https://stackoverflow.com/a/3679918

It's not working for me anymore either however...

**Comment by mdmurbach** [comment]
Not sure if we can do anything about the black background :p

**Comment by BGerwe** [comment]
I think this might help? https://stackoverflow.com/questions/55401706/add-background-solid-fill-to-altair-graph

I would add it in myself, but I'm out of town doing these reviews on my phone.

**Comment by mdmurbach** [comment]
Added a white background to the chart by default. Thanks for such a thorough review while traveling :) I'll take a deeper dive into the scale argument issue tonight and either (i) remove it since it doesn't do anything and re-open the old issue or (ii) hopefully, fix it :joy:

File: impedance/plotting.py (hunk 1 above)
Comment (mdmurbach) [reply impedance/plotting.py]: Looks like it was a recent change in matplotlib that [added a new way of doing this](https://matplotlib.org/users/prev_whats_new/whats_new_3.0.html#ability-to-scale-axis-by-a-fixed-order-of-magnitude). Think it should work now (although I'm still not sure this is the best way of handling widely varying impedances) :smile:

**Comment by dt-schwartz** [comment]
Wow, nice polishing work!

**Comment by mdmurbach** [comment]
Thanks to you too for the good review, @BGerwe. :rocket: Looks good to go!

@dt-schwartz, thanks! Things are really coming together. Will be exciting to finally release a v1.0 and write up a paper in JOSS soon!

**Pull request merged**